processing/src/main/java/io/druid/segment/StringDimensionMergerV9.java

```diff
@@ -196,6 +200,20 @@ public void writeMergedValueMetadata(List<IndexableAdapter> adapters) throws IOE
     setupEncodedValueWriter();
   }
 
+  protected List<Indexed<String>> toDimValuesList(List<IndexableAdapter> adapters) throws IOException
```

Comment:
As far as I can understand you changed this class in order to be able to extend it in your code. See comment to `SmooshedFileMapper`, don't do this. Copy class entirely into your codebase.